airflow-core/src/airflow/ui/src/components/Assets/AssetEventsFilter.tsx

```diff
@@ -0,0 +1,102 @@
+
+import { DateTimeInput } from "src/components/DateTimeInput";
+import { SearchBar } from "src/components/SearchBar";
+import { SearchParamsKeys } from "src/constants/searchParams";
+
+export const AssetEventsFilter = () => {
+  const { t: translate } = useTranslation("common");
+  const { DAG_ID, END_DATE, START_DATE, TASK_ID } = SearchParamsKeys;
+  const [searchParams, setSearchParams] = useSearchParams();
+  const startDate = searchParams.get(START_DATE) ?? "";
+  const endDate = searchParams.get(END_DATE) ?? "";
+  const dagId = searchParams.get(DAG_ID) ?? "";
+  const taskId = searchParams.get(TASK_ID) ?? "";
+  const [resetKey, setResetKey] = useState(0);
+  const handleFilterChange = useCallback(
+    (paramKey: string) => (value: string) => {
+      if (value === "") {
+        searchParams.delete(paramKey);
+      } else {
+        searchParams.set(paramKey, value);
+      }
+      setSearchParams(searchParams);
+    },
+    [searchParams, setSearchParams],
+  );
+  const handleResetFilters = useCallback(() => {
+    searchParams.delete(START_DATE);
+    searchParams.delete(END_DATE);
+    searchParams.delete(DAG_ID);
+    searchParams.delete(TASK_ID);
+    setSearchParams(searchParams);
+    setResetKey((prev) => prev + 1);
+  }, [searchParams, setSearchParams, START_DATE, END_DATE, DAG_ID, TASK_ID]);
+
+  return (
+    <VStack align="start" gap={4} paddingY="4px">
+      <HStack flexWrap="wrap" gap={4}>
+        <Box w="200px">
+          <Text fontSize="xs">{translate("common:table.from")}</Text>
+          <DateTimeInput
+            onChange={(event) => handleFilterChange(START_DATE)(event.target.value)}
+            value={startDate}
+          />
+        </Box>
+        <Box w="200px">
+          <Text fontSize="xs">{translate("common:table.to")}</Text>
+          <DateTimeInput
+            onChange={(event) => handleFilterChange(END_DATE)(event.target.value)}
+            value={endDate}
+          />
+        </Box>
+        <Box w="200px">
+          <Text fontSize="xs">{translate("common:filters.dagDisplayNamePlaceholder")}</Text>
+          <SearchBar
+            defaultValue={dagId}
+            hideAdvanced
+            hotkeyDisabled={true}
+            key={`dag-id-${resetKey}`}
+            onChange={handleFilterChange(DAG_ID)}
+            placeHolder={translate("common:filters.dagDisplayNamePlaceholder")}
+          />
+        </Box>
+        <Box w="200px">
+          <Text fontSize="xs">{translate("common:filters.taskIdPlaceholder")}</Text>
+          <SearchBar
+            defaultValue={taskId}
+            hideAdvanced
+            hotkeyDisabled={true}
+            key={`task-id-${resetKey}`}
+            onChange={handleFilterChange(TASK_ID)}
+            placeHolder={translate("common:filters.taskIdPlaceholder")}
+          />
+        </Box>
+        <Button onClick={handleResetFilters} w="200px">
+          {translate("common:table.filterReset_other")}
+        </Button>
```

Comment:
Lets put reset button in a box for better styling, and add `filterCount` for conditional reset button (see X`ComFilters.tsx`:138)

```suggestion
        <Box alignSelf="end">
          {filterCount > 0 && (
            <Button onClick={handleResetFilters} size="md" variant="outline">
              <LuX />
              {translate("common:table.filterReset", { count: filterCount })}
            </Button>
          )}
        </Box>
```

Reply:
before
<img width="645" height="259" alt="CleanShot 2025-09-04 at 11 15 08" src="https://github.com/user-attachments/assets/c8c8418a-148e-4887-8647-00dc4f3f0462" />
after
<img width="645" height="259" alt="CleanShot 2025-09-04 at 11 14 19" src="https://github.com/user-attachments/assets/45683d86-61cb-4154-b624-d08a2b0e16f1" />

Reply:
Thank you for the very useful suggestion. I've committed them.